tests/expressions/tests.py

```diff
@@ -584,6 +584,28 @@ def test_outerref_with_operator(self):
         outer = Company.objects.filter(pk__in=Subquery(inner.values('pk')))
         self.assertEqual(outer.get().name, 'Test GmbH')
 
+    def test_exists_in_filter(self):
+        inner = Company.objects.filter(ceo=OuterRef('pk')).values('pk')
+        outer_1 = Employee.objects.filter(Exists(inner))
+        outer_2 = Employee.objects.annotate(found=Exists(inner)).filter(found=True)
+        self.assertCountEqual(outer_2, outer_1)
+        self.assertFalse(Employee.objects.exclude(Exists(inner)).exists())
+        self.assertCountEqual(outer_2, Employee.objects.exclude(~Exists(inner)))
+
+    def test_case_valid_in_filter_if_boolean_output_field(self):
+        is_ceo = Company.objects.filter(ceo=OuterRef('pk'))
+        is_poc = Company.objects.filter(point_of_contact=OuterRef('pk'))
+        outer_1 = Employee.objects.filter(Case(
+            When(Exists(is_ceo), then=Value(True)),
+            When(Exists(is_poc), then=Value(True)),
+            default=Value(False, output_field=models.BooleanField())
+        ))
```

Comment:
I'd be great if we'd be able to express this as `Q(Exists(is_ceo)) | Q(Exists(is_poc))` (or even `Exists(is_ceo) | Exists(is_poc)` in the future) instead. Actually it's possible the `Q()` approach already works?

Reply:
FWIW `Q(Exists(is_ceo)) | Q(Exists(is_poc))` already works and `Exists(is_ceo) | Exists(is_poc)` doesn't require much changes.

```diff
diff --git a/django/db/models/expressions.py b/django/db/models/expressions.py
index 5b82ae97a7..d18de75e9a 100644
--- a/django/db/models/expressions.py
+++ b/django/db/models/expressions.py
@@ -101,6 +101,8 @@ class Combinable:
         return self._combine(other, self.BITRIGHTSHIFT, False)

     def __or__(self, other):
+        if getattr(self, 'conditional', False) and getattr(other, 'conditional', False):
+            return Q(self) | Q(other)
         raise NotImplementedError(
             "Use .bitand() and .bitor() for bitwise logical operations."
         )
diff --git a/tests/expressions/tests.py b/tests/expressions/tests.py
index 6c00f813d9..d3d75f1ce1 100644
--- a/tests/expressions/tests.py
+++ b/tests/expressions/tests.py
@@ -595,11 +595,9 @@ class BasicExpressionsTests(TestCase):
     def test_case_valid_in_filter_if_boolean_output_field(self):
         is_ceo = Company.objects.filter(ceo=OuterRef('pk'))
         is_poc = Company.objects.filter(point_of_contact=OuterRef('pk'))
-        outer_1 = Employee.objects.filter(Case(
-            When(Exists(is_ceo), then=Value(True)),
-            When(Exists(is_poc), then=Value(True)),
-            default=Value(False, output_field=models.BooleanField())
-        ))
+        outer_1 = Employee.objects.filter(
+            Exists(is_ceo) | Exists(is_poc)
+        )
         self.assertQuerysetEqual(
             outer_1,
             ['<Employee: Joe Smith>', '<Employee: Frank Meyer>', '<Employee: Max Mustermann>'],
```

Reply:
Oh, that's really nice.